rotkehlchen/exchanges/binance.py

```diff
@@ -530,6 +537,66 @@ def _query_spot_balances(
 
         return balances
 
+    def _query_funding_balances(
+            self,
+            balances: defaultdict[AssetWithOracles, Balance],
+    ) -> defaultdict[AssetWithOracles, Balance]:
+        funding_balances = self.api_query_list('sapi', 'asset/get-funding-asset', request_type='post')  # noqa: E501
+        if len(funding_balances) == 0:
+            return balances
+
+        for entry in funding_balances:
+            try:
+                asset_symbol = str(entry['asset'])
+                free = deserialize_asset_amount(entry['free'])
+                locked = deserialize_asset_amount(entry['locked'])
+            except KeyError as e:
+                raise RemoteError(f'Binance funding balance asset entry did not contain key {e!s}') from e  # noqa: E501
+            except DeserializationError as e:
+                raise RemoteError('Failed to deserialize an amount from binance funding balance asset entry') from e  # noqa: E501
+
+            amount = free + locked
+            if amount == ZERO:
+                continue
+
+            try:
+                asset = asset_from_binance(asset_symbol)
+            except UnsupportedAsset as e:
+                if e.identifier != 'ETF':
+                    log.error(
+                        f'Found unsupported {self.name} asset {e.identifier}. '
+                        f'Ignoring its balance query.',
+                    )
+                continue
+            except UnknownAsset as e:
+                log.error(
+                    f'Found unknown {self.name} asset {e.identifier}. '
+                    f'Ignoring its balance query.',
+                )
+                continue
+            except DeserializationError:
+                log.error(
+                    f'Found {self.name} asset with non-string type {type(entry["asset"])}. '
+                    f'Ignoring its balance query.',
+                )
+                continue
```

Comment:
this one can only be raised if the symbol is not a string but if you control the type I believe it is safe to remove this

Reply:
This one was resolveed but the exception is there and no comment was provided

Reply:
Now we're processing and error handling the raw data coming from the API, so kept that part.